### PR TITLE
Added a purchase history 

### PR DESCRIPTION
**Pull request by Superdoude1999**
Added a purchase history to see previously bought items. It is located in the same place as the Sold items, the fragment now serves to display both bought and sold items.

### CODE REVIEW

File: app/src/main/java/com/example/sharingang/ui/fragments/DetailedItemFragment.kt [hunk 1]
```diff
@@ -223,19 +223,19 @@ class DetailedItemFragment : Fragment() {
...
             Toast.makeText(context, getString(R.string.incorrect_quantity), Toast.LENGTH_SHORT).show()
-            binding.buyButton.isEnabled = true
         } else {
             lifecycleScope.launch {
-                if (paymentProvider.requestPayment(args.item, quantity)) {
-                    itemViewModel.sellItem(args.item)
-                    val newQuantity = args.item.quantity - quantity
+                binding.buyButton.isEnabled = true
+                if (paymentProvider.requestPayment(boughtItem, quantity)) {
+                    itemViewModel.sellItem(boughtItem)
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/ui/fragments/DetailedItemFragment.kt:233]: I think you should only set it to sold after the new quantity has been updated.

File: app/src/main/java/com/example/sharingang/viewmodels/ItemsViewModel.kt [hunk 2]
```diff
@@ -69,9 +76,7 @@ class ItemsViewModel @Inject constructor(
     val userItemsAndRequests: LiveData<List<Item>>
         get() = _userItemsAndRequests
 
-    private val _userSoldItems = MutableLiveData<List<Item>>()
-    val userSoldItems: LiveData<List<Item>>
-        get() = _userSoldItems
+//
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/viewmodels/ItemsViewModel.kt:79]: I think you can remove this comment here.

**Review by bafavre — APPROVED** [review]
Nice, well done! Just a few remarks :)

File: app/src/main/java/com/example/sharingang/viewmodels/ItemsViewModel.kt (hunk 2 above)
Comment (Superdoude1999) [reply app/src/main/java/com/example/sharingang/viewmodels/ItemsViewModel.kt]: Thanks I removed it

File: app/src/main/java/com/example/sharingang/ui/fragments/DetailedItemFragment.kt (hunk 1 above)
Comment (Superdoude1999) [reply app/src/main/java/com/example/sharingang/ui/fragments/DetailedItemFragment.kt]: Yeah turns out I didn't need that function in the first place so I just removed it, it was from old code.

**Review by lumapools — APPROVED** [review]
Well done, looking good! I'm happy you used the custom radio buttons! Also thank you for providing pictures ❤️

**Pull request merged**